File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = []
        for i in range(len(nums)):
            product = 1
            for n in range(len(nums)):
                if n == i:
                    continue
                else:
                    product = product * nums[n]
            output.append(product)
        return output
            

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        while (l <= r) {
            int half = l + (r - l) / 2;
            if (target == nums[half]) {
                return half;
            }
            if (target > nums[half]) {
                l = half + 1;
            } else {
                r = half - 1;
            }
        }
        return -1;
        
    }
};
